Add documentation for configuring open forms with appointments (#683)
open-formulieren/open-forms: docs/configuration/appointment.rst

.. _configuration_appointment:

=========================
Appointment configuration
=========================

> Appointments can be made using two separate services.  JCC and Qmatic.

Reviewer: Can you split up JCC and Qmatic in separate files, in a folder `appointment` similar to the other plugins: https://github.com/open-formulieren/open-forms/tree/master/docs/configuration

This was changed recently!

Open Forms can be configured to make use of one of these services.

JCC
---

1. In Open Forms navigate to: **Configuration** > **JCC configuration**
2. Click the **Green Plus Button** and fill in the following details:

   * **Label**: JCC
   * **Ontvangende organisatie**: Open Forms
   * **Ontvangende applicatie**: Open Forms
   * **Versturende organisatie**: Horstaandemaas
   * **Versturende applicatie**: Afspraak Acceptatie

>    * **URL**: https://afspraakacceptatie.horstaandemaas.nl/JCC/Horst%20aan%20de%20Maas/WARP/GGS2/GenericGuidanceSystem2.asmx?wsdl

Author: This is what I've used for testing JCC but I'm not sure if we want to mention something different here?  Also for this particular endpoint I don't think the Ontvangende/Versturende fields actually matter since I tried a few different values with testing this and everything seemed to work.

Reviewer: Stop, remove that URL please!

Please use: `https://example.com/jcc/GenericGuidanceSystem2.asmx?wsdl`

Reviewer: Stop, remove that URL please and rebase as if it was never here!

Please use: `https://example.com/jcc/GenericGuidanceSystem2.asmx?wsdl`

3. Click **Save**

Qmatic
------

> 

Reviewer: ```suggestion

This plugin is made for the *Qmatic Orchestra Calendar Public Appointment API* and is currently untested.

```

> 1. Obtain a `Qmatic API key`_ from Qmatic.

Reviewer: ```suggestion
1. You will need to have a contract with `Qmatic`_ to use this plugin.
```

2. In Open Forms navigate to: **Configuration** > **Qmatic configuration**
3. Click the **Green Plus Button** and fill in the following details:

   * **Label**: Qmatic
   * **Type**: ORC (Overige)

>    * **API root URL**: ``TODO: FILL ME IN``

Reviewer: ```suggestion
   * **API root URL**: *The API endpoint provided by Qmatic*
```

   * **Authorization type**: API key
   * **Header key**: X-Api-Key

>    * **Header value**: *The Qmatic API key obtained in step 1*

Reviewer: ```suggestion
   * **Header value**: *The API-key provided by Qmatic*
```

>    * **OAS**: ``TODO: FILL ME IN``

Author: Can you verify this is correct and add the correct urls for the `API root URL` and `OAS`.

Reviewer: No clue, and the they differ I think per organisation.

>    * **OAS**: ``TODO: FILL ME IN``

Reviewer: ```suggestion
   * **OAS**: *URL to the Open API-specification provided by Qmatic*
```

4. Click **Save**

Appointments config
-------------------

Regardless of which service you decide to use navigate
to **Configuration** > **Appointments config** and select which service you set up
from the dropdown.

> The Appointments configuration is now completed.

Reviewer: This part can be moved below the tree listing the plugins.

> .. _`Qmatic API key`: https://www.qmatic.com/

Author: Is this the correct url to redirect someone to in order to get an API key?  I'm guessing there is a more direct url we could send the users to.

Reviewer: Qmatic is not a service where you can just get an API key, you need a contract with them to use their cloud service.

> .. _`Qmatic API key`: https://www.qmatic.com/

Reviewer: ```suggestion
.. _`Qmatic`: https://www.qmatic.com/solutions/online-appointment-booking/
```
